plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/connection/ConnectionWizardPageModel.java

```diff
@@ -348,9 +348,12 @@ public IStatus connect() {
 		IStatus status = Status.OK_STATUS;
 		listener.secureStoreException = null;
 		try {
-			IConnection connection = createConnection(connectionFactory, connectionAuthenticationProvider);
+			IConnection connection = 
+					isNewConnection()
+					? createConnection(connectionFactory, connectionAuthenticationProvider)
```

Comment:
we should never edit/use the (edited) connection directly since you can cancel the wizard. You expect the edited connection to be untouched if you cancel the wizard.